src/VisualStudio/Core/Def/ValueTracking/ValueTrackingToolWindow.cs

```diff
@@ -49,28 +62,33 @@ public ValueTrackingToolWindow() : base(null)
             Content = _root;
         }
 
-        public ValueTrackingToolWindow(ValueTrackingTreeViewModel viewModel)
-            : base(null)
+        public void Initialize(ValueTrackingTreeViewModel viewModel, Workspace workspace, IThreadingContext threadingContext)
         {
-            Caption = ServicesVSResources.Value_Tracking;
-            Content = _root;
+            Contract.ThrowIfTrue(Initialized);
+
+            Initialized = true;
             ViewModel = viewModel;
+            _workspace = workspace;
+            _threadingContext = threadingContext;
+
+            _workspace.WorkspaceChanged += OnWorkspaceChanged;
         }
 
-        public TreeItemViewModel? Root
+        private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
         {
-            get => ViewModel?.Roots.Single();
-            set
-            {
-                if (value is null)
-                {
-                    return;
-                }
+            Contract.ThrowIfFalse(Initialized);
 
-                Contract.ThrowIfNull(ViewModel);
+            if (e.Kind is WorkspaceChangeKind.SolutionCleared
+                       or WorkspaceChangeKind.SolutionRemoved)
+            {
+                RoslynDebug.AssertNotNull(_threadingContext);
+                RoslynDebug.AssertNotNull(ViewModel);
 
-                ViewModel.Roots.Clear();
-                ViewModel.Roots.Add(value);
+                _ = _threadingContext.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    ViewModel.Roots.Clear();
+                });
```

Comment:
Do you want to wrap this with an async waiter for integration tests?

Reply:
afaik we don't have any integration tests for clearing solution/closing solution. If we add we definitely would need to